onnxruntime/core/providers/dml/OperatorAuthorHelper/OperatorHelper.cpp

```diff
@@ -2480,6 +2481,50 @@ namespace OperatorHelper
                 MLOperatorTensor outputSizesTensor = kernelInformation.GetConstantInputTensor(3);
                 ReadCpuLocalTensorIntoInt32(outputSizesTensor, /*out*/ outputSizes);
             }
+
+            axes = kernelInformation.GetAttributes().GetOptionalAttributeVectorInt32(AttrName::Axes);
+            // Handle possible axes input
+            if (opsetVersion >= 18 && !axes.empty())
+            {
+                uint32_t dimCount = gsl::narrow_cast<uint32_t>(m_inputDimensions.size());
+                HandleEmptyAxes(axes, m_inputDimensions, false);
+                HandleNegativeAxes(axes, dimCount);
+
+                // Taken from https://github.com/onnx/onnx/blob/3d69db8fd16873d68e7033479467f9478562a12d/onnx/reference/ops/op_resize.py#L303
+                if (!m_scales.empty())
+                {
+                    std::vector<float> newScales(dimCount, 1.f);
+                    // Fill in roi and scales/sizes
+                    uint32_t numAxes = gsl::narrow_cast<uint32_t>(axes.size());
+                    for (int32_t i = 0; i < axes.size(); i++)
```

Comment:
Avoid mixing different types in the counter vs the loop limit. If you ever find yourself using `int` in a loop, then double check whether that should really be a `uint32_t` or a `size_t`.